clm_demeter/prepare_baselayer.py

```diff
@@ -0,0 +1,161 @@
+    
+    """
+    # Join the subset grid coordinate data with full coordinate data
+    coarse_grid_data = pd.merge(coarse_grid_coord, data, how='left')
+
+    # Interpolate from coarse grid to finer resolution grid. 
+    # Convert dataframe to ndarray for method='nearest' to work
+    data_fine = griddata(points=coarse_grid_data[['Latcoord','Loncoord']].values, 
+                         values=coarse_grid_data[data.columns[2]].values, 
+                         xi=fine_grid_coord.values, 
+                         method='nearest')
+    
+    return data_fine
+# ------------------------------------------------------------------------
+
+project_dir = '/pic/projects/im3/lulc_thrust_area/minchen/Demeter_0.05deg_trial/Demeter_0.05deg_trial'
+
+# Base layer file for Plant Function Type (PFT)
+# baselayer_pft_file = os.path.join(project_dir, 'mksrf_78pft_landuse_rc2000_c130927.nc')
+baselayer_pft_file = 'mksrf_78pft_landuse_rc2000_c130927.nc'
+
+# Open netCDF file
+dataset = Dataset(baselayer_pft_file, mode='r') # file handle, open in read only mode
+
+# Read latitude, longitude values
+lat = dataset.variables['LAT']  # dim = lat
+lon = dataset.variables['LON']  # dim = lon
+
+lat_xy = dataset.variables['LATIXY'] # dim = (lat, lon)
+lon_xy = dataset.variables['LONGXY'] # dim = (lat, lon)
+landmask = dataset.variables['LANDMASK'] # dim = (lat, lon)
+
+# Read data for PFT variable [pft (79), lat (3600), lon (7200)]
+pft = dataset.variables['PCT_PFT']
+
+close the file
+dataset.close()
+
+# flip the data in latitude so North Hemisphere is up on the plot
+# flipud will convert the netcdf variable to numpy array
+lat      = np.flipud(lat) 
+lat_xy   = np.flipud(lat_xy)
+lon_xy   = np.flipud(lon_xy)
+landmask = np.flipud(landmask)
+pft      = pft[:,::-1,:] 
+
+# PFT values are converted from percentage (0-100) to fraction (0-1)
+pft = pft*0.05*0.05*0.01; # unit conversion
+
+# Generating land see mask
+# pft_tot = np.sum(pft, axis=0) # Sum values for all PFTs
+# landmask = (pft_tot > 0)      # identify land grid cells
+landmask[(lat <= -55),:] = 0;   # assigning 0 to Antartica [lat less than -55] grid cells
+
+# Reshape to convert lat lon 2D array values to 1 dimension
+pft = np.reshape(pft, (79,-1))      # from [79, 3600, 7200] to [79, 3600x7200]
+lat_xy = np.reshape(lat_xy, -1)     # from [3600, 7200] to [3600x7200, ]
+lon_xy = np.reshape(lon_xy, -1)     # from [3600, 7200] to [3600x7200, ]
+landmask = np.reshape(landmask, -1) # from [3600, 7200] to [3600x7200, ]
+
+# Swap axes for pft array from [79, 3600x7200] to [3600x7200, 79]
+pft = np.swapaxes(pft, 0, 1)
+
+# Round lat and lon values to 3 decimal digits
+lat_xy = lat_xy.round(3)
+lon_xy = lon_xy.round(3)
+
+# Generate FID for various grid cells (range from 0:3600x7200)
+fid = np.array(range(np.shape(lat)[0] * np.shape(lon)[0]))
+
+# Generate PFT column names
+PFT_names = ['PFT'+ str(x) for x in range(1,pft.shape[1]+1)]
```

Comment:
For Python, need space between commas for arguments (e.g. `range(1, pft.shape[1] + 1)]`